app/src/main/java/fi/thl/koronahaavi/common/ENEnablerFragment.kt

```diff
@@ -136,10 +137,25 @@ fun Context.showEnableFailureReasonDialog(error: EnableENError) {
         is EnableENError.MissingCapability -> {
             builder.setMessage(R.string.enable_err_missing_capability)
         }
+
+        is EnableENError.ApiNotSupported -> {
+            val apiError = error.enApiError?.let {
+                when (it) {
+                    is ENApiError.DeviceNotSupported -> getString(R.string.enable_err_api_connection_device)
+                    is ENApiError.AppNotAuthorized -> getString(R.string.enable_err_api_connection_unauthorized)
+                    is ENApiError.Failed -> getString(R.string.enable_err_api_connection_generic, it.errorCode ?: 0)
+                }
+            }
+
+            builder.setMessage(
+                getString(R.string.enable_err_api_not_supported, apiError)
+            )
+        }
+
         is EnableENError.Failed -> {
             builder.setMessage(
                 error.code?.let {
-                    getString(R.string.enable_error_api_code, it)
+                    getString(R.string.enable_error_api_code, it, error.connectionErrorCode ?: 0)
```

Comment:
Unless I'm mistaken connectionErrorCode originates from APIException.status.getStatusCode() which should be one of ExposureNotificationStatusCodes which extends CommonStatusCodes and that defines 0 as a success value (like it is often used). Therefore I'd suggest to use some other default value here just in case - something that isn't one of the documented values, e.g. 0x54 48 4c (=THL) :D

Reply:
Good point, just changed those to show null if not defined (normally they should be defined)